tests/test-cong.cpp

```diff
@@ -538,6 +538,9 @@ namespace libsemigroups {
     // non-trivial classes.
     // REQUIRE(cong.number_of_non_trivial_classes() == 0);
     REQUIRE(cong.number_of_classes() == POSITIVE_INFINITY);
+    // cong hasn't started yet, because the presentation is obviously infinite.
+    REQUIRE(!cong.started());
+    cong.run();
```

Comment:
Prior to this PR, calling `Congruence::number_of_classes` would always invoke `Congruence::run`. However, with the changes in this PR, there are instances where this does not happen (namely when the given presentation is obviously infinite). As such, some of the tests that required `Congruence::run` to have been called began to fail.

I addressed this by directly calling `Congruence::run` but otherwise left everything else unchanged. Perhaps it would be better to restructure the tests. I'm open to any opinions on this one.